bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,295 @@
+                    // Disconnected from Qbus Server, try to reconnect
+                    logger.info("No connection with Qbus Server. Try to restart.");
+                    comm.restartCommunication();
+                    if (!comm.communicationActive()) {
+                        bridgeOffline("No connection with Qbus Server");
+                        return;
+                    }
+
+                } else {
+
+                    // Controller disconnected from Qbus client, try to reconnect controller
+                    if (!comm.clientConnected()) {
+                        logger.info("No connection with Qbus Client. Try to restart.");
+                        comm.restartCommunication();
+                        if (!comm.clientConnected()) {
+                            bridgeOffline("No connection with Qbus Client");
+                            return;
+                        }
+                    }
+                }
+            }
+
+            logger.info("Connection with Qbus Server & Client is still active.");
+            updateStatus(ThingStatus.ONLINE);
+
+        }, refreshInterval, refreshInterval, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Disposes the Bridge and stops communication with the Qbus server
+     */
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+        }
+
+        refreshTimer = null;
+
+        QbusCommunication comm = getCommunication();
+        if (comm != null) {
+            comm.stopCommunication();
+        }
+
+        comm = null;
+    }
+
+    /**
+     * Update the configuration parameters
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        Configuration configuration = editConfiguration();
+        for (Entry<String, Object> configurationParmeter : configurationParameters.entrySet()) {
+            configuration.put(configurationParmeter.getKey(), configurationParmeter.getValue());
+        }
+        updateConfiguration(configuration);
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * Sets the configuration prameters
+     */
+    protected synchronized void setConfig() {
+        config = getConfig().as(QbusConfiguration.class);
+    }
+
+    /**
+     * Get the Qbus communication object.
+     *
+     * @return Qbus communication object
+     */
+    public @Nullable QbusCommunication getCommunication() {
+        return this.qbusComm;
+    }
+
+    /**
+     * Get the IP-address of the Qbus server.
+     *
+     * @return the addr
+     */
+
+    @SuppressWarnings("null")
+    public @Nullable InetAddress getAddr() {
+        InetAddress addr = null;
+        try {
+            addr = InetAddress.getByName(config.addr);
+        } catch (UnknownHostException e) {
+            logger.debug("Cannot resolve hostname {} to IP adress", config.addr);
+        }
+        return addr;
+    }
```

Comment:
You could throw the `UnknownHostException` and handle it in `initialize()`. That would make the null suppression unnecessary. In fact that would make the whole method unnecessary. It's better to throw an exception in case of an error instead of returning null. This way you have the error message in the calling method and can integrate it in the message the user gets presented.